storer/files.go

```diff
@@ -50,11 +59,45 @@ func GetTaskFileWriteCloser(id int) (io.WriteCloser, error) {
 			log.Printf("Cannot create directory %s Error %s", dir, err)
 		}
 	}
-	file, err := os.Create(fmt.Sprintf("%s/task-%d", dir, id))
+	file, err := os.Create(getTaskPath(dir, id))
 	if err != nil {
 		log.Printf("Cannot create file task-%d Error %s", id, err)
 		return nil, err
 
 	}
 	return file, nil
 }
+
+func S3UploadTask(bucket string, id int) {
+	dir := viper.GetString(misc.OutDirKey)
+	awsRegion := viper.GetString(misc.AWSRegion)
+	filename := getTaskPath(dir, id)
+	file, err := os.Open(filename)
+	if err != nil {
+		fmt.Println("Failed to open file for upload to S3 bucket", filename, err)
+		os.Exit(1)
+	}
+	defer file.Close()
+
+	//TODO fix credentials
+	conf := aws.Config{
+		Region:      aws.String(awsRegion),
+		Credentials: credentials.NewSharedCredentials("", "production_42"),
+	}
+	sess := session.New(&conf)
+	svc := s3manager.NewUploader(sess)
+	if viper.GetBool(misc.DebugKey) {
+		fmt.Println("Uploading file to S3")
+	}
+	result, err := svc.Upload(&s3manager.UploadInput{
+		Bucket: aws.String(bucket),
+		Key:    aws.String(filepath.Base(filename)),
+		Body:   file,
+	})
+	if err != nil {
+		fmt.Println("error", err)
+		os.Exit(1)
+	}
+
+	fmt.Printf("Successfully uploaded %s to %s\n", filename, result.Location)
```

Comment:
I'd rather use here log.Printf instead of fmt.Printf. Because log prints to std err.